File: tests/fixtures/check_then_act/ts/local_collection_fixture.ts
// Collections built inside the function are not shared state.
export async function gather(repo: any, ids: string[]) {
  const out: string[] = [];
  const seen = new Map<string, boolean>();
  for (const id of ids) {
    const value = repo.get(id);
    if (!value) continue;
    await repo.touch(id);
    out.push(value);
    seen.set(id, true);
  }
  return out;
}

File: tests/fixtures/check_then_act/py/local_collection_fixture.py
# Collections built inside the function are not shared state.
async def gather(repo, ids):
    out = []
    seen = {}
    for item_id in ids:
        value = repo.get(item_id)
        if not value:
            continue
        await repo.touch(item_id)
        out.append(value)
        seen.update({item_id: True})
    return out
